my_account.rb

```diff
@@ -46,8 +48,56 @@
 helpers StyledFlash
 
 enable :sessions
+set :session_secret, ENV['RACK_COOKIE_SECRET'] 
 set server: 'thin', connections: []
 
+use OmniAuth::Builder do
+  provider :openid_connect, {
+    issuer: 'https://weblogin.lib.umich.edu',
+    discovery: true,
+    client_auth_method: 'jwks',
+    scope: [:openid, :profile, :email],
+    client_options: {
+      identifier: 'patron-account-testing',
+      secret: ENV['WEBLOGIN_SECRET'],
+      redirect_uri: "#{ENV['PATRON_ACCOUNT_BASE_URL']}/auth/openid_connect/callback"
+    }
+  }
+end
+
+get '/auth/openid_connect/callback' do
+  auth = request.env['omniauth.auth']
+  info = auth[:info]
+  session[:authenticated] = true
+  session[:uniqname] = info[:nickname]
+  session[:expires_at] = Time.now + auth.credentials.expires_in
```

Comment:
For anything that is persisted, I prefer to use UTC to eliminate any timezone oddities.